src/app/components/messages/messages.component.ts

```diff
@@ -13,25 +12,24 @@ import {
   Output,
   QueryList,
   ViewChildren,
-  WritableSignal,
-  computed,
-  signal,
 } from '@angular/core'
 import { ReactiveFormsModule } from '@angular/forms'
+import { MatButtonModule } from '@angular/material/button'
 import { MatDividerModule } from '@angular/material/divider'
 import { MatFormFieldModule } from '@angular/material/form-field'
 import { MatIconModule } from '@angular/material/icon'
 import { MatInputModule } from '@angular/material/input'
 import { MatTooltipModule } from '@angular/material/tooltip'
-import { Message_Status, Translation } from '@buf/expectdigital_translate-agent.bufbuild_es/translate/v1/translate_pb'
-import { Subscription } from 'rxjs'
-
-import { MatButtonModule } from '@angular/material/button'
-import { MatSnackBar } from '@angular/material/snack-bar'
+import { Message, Message_Status } from '@buf/expectdigital_translate-agent.bufbuild_es/translate/v1/translate_pb'
 import hljs from 'highlight.js'
+import { Subscription } from 'rxjs'
 import { slideInOut } from 'src/app/animation'
 import messageFormat2 from 'src/app/highlight'
-import { TranslateClientService } from 'src/app/services/translate-client.service'
+
+export interface EmittedData {
+  event: Event
```

Comment:
If you know the DOM element, specify it here. Avoid letting data consumers figure it out.

```suggestion
  event: Event & { element: HTMLTextAreaElement }
```

Reply:
Is only "(event.target as HTMLTextAreaElement).innerText" consumed by consumers? Is there a better way than exposing the DOM element? Provide the value the consumer is interested in.